Code/Framework/AzToolsFramework/AzToolsFramework/AssetEditor/AssetEditorWidget.cpp

```diff
@@ -340,6 +340,10 @@ namespace AzToolsFramework
             auto serializeContext = AZ::EntityUtils::GetApplicationSerializeContext();
             serializeContext->CloneObjectInplace((*m_inMemoryAsset.GetData()), asset.GetData());
 
+            // Make sure the saved state key is reset since this could be a file opened with the same property editor isntance
+            m_savedStateKey = AZ::Crc32();
+            m_savedStateKey.Add(&asset.GetId(), sizeof(AZ::Data::AssetId));
```

Comment:
Instead of having two lines to reset the m_savedStateKey and caluculate a new Crc32, this can just calculate the Crc32 using the AZ::Crc32 constructor which accepts a buffer and a size.
```suggestion
            m_savedStateKey = AZ::Crc32(&asset.GetId(), sizeof(AZ::Data::AssetId);
```